misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_promote.sql

```diff
@@ -88,6 +88,26 @@
     {{ deploy_await(poll_interval) }}
 {% endif %}
 
+{% call statement('alter_sinks', fetch_result=True, auto_begin=True) -%}
+-- 'ALTER SINK ... SET FROM ...' cannot be run inside a transaction block
+{% for schema in schemas %}
+    {% set sinks_and_upstream_relations = get_sinks_and_upstream_relations(schema) %}
+    {% if sinks_and_upstream_relations is not none and sinks_and_upstream_relations.rows %}
+        {% for sink in sinks_and_upstream_relations.rows %}
+            {% set sink_name = adapter.quote(sink[2]) %}
+            {% set sink_schema_name = adapter.quote(sink[3]) %}
+            {% set sink_database_name = adapter.quote(sink[4]) %}
+            {% set upstream_relation_name = adapter.quote(sink[5]) %}
+            {% set upstream_relation_schema = adapter.quote(sink[6]) %}
+            {% set deploy_schema = adapter.quote(sink[6] ~ "_dbt_deploy") %}
+            {% set new_upstream_relation = get_current_database() ~ '.' ~ deploy_schema ~ '.' ~ upstream_relation_name %}
+            {{ log("Altering sink " ~ sink_database_name ~ "." ~ sink_schema_name ~ "." ~ sink_name ~ " to use new upstream relation " ~ new_upstream_relation, info=True) }}
+            ALTER SINK {{ sink_database_name ~ "." ~ sink_schema_name ~ "." ~ sink_name }} SET FROM {{ new_upstream_relation }};
```

Comment:
Ah, yeah; this separation makes sense to me!

Reply:
Reading through the draft for the `ALTER SINK` documentation, I'm slightly concerned about this scenario:

> When using the Avro format with a schema registry care, the generated Avro
schema for the new relation must be compatible with the previously published
schema. If that's not the case, the `ALTER SINK` command will succeed, but the
subsequent execution of the sink will result in errors and will not be able to
make progress.

Is there a way to validate that the sink is healthy? Also, shouldn't the swap operation happen _before_ `ALTER SINK` is executed?

Reply:
Oh, nevermind.

Reply:
They need to be continually monitoring the sink for errors to notice the failure to publish the schema. Agree it’s a poor UX. I think the right place to handle that check is in ALTER SINK itself though. Discussing more here! https://github.com/MaterializeInc/materialize/issues/23794#issuecomment-2162049350

Reply:
That was my first thought, too!